### PR TITLE
refactor(connector): [Payeezy] remove default case handling

### PR DESCRIPTION
**Pull request by HeetVekariya**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- Fix issue #2281
- Changes to be made : Remove default case handling
- File changed: ```transformers.rs``` (crates/router/src/connector/payeezy/transformers.rs)

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

No test cases. As In this PR only error message have been populated for all default cases.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by AkshayaFoiger — APPROVED** [review]

File: crates/router/src/connector/payeezy/transformers.rs [hunk 1]
```diff
@@ -37,11 +37,14 @@ impl TryFrom<utils::CardIssuer> for PayeezyCardType {
...
             utils::CardIssuer::Visa => Ok(Self::Visa),
-            _ => Err(errors::ConnectorError::NotSupported {
-                message: issuer.to_string(),
-                connector: "Payeezy",
+
+            utils::CardIssuer::Maestro | utils::CardIssuer::DinersClub | utils::CardIssuer::JCB => {
+                Err(errors::ConnectorError::NotSupported {
+                    message: issuer.to_string(),
+                    connector: "Payeezy",
+                }
```
Comment (srujanchikke) [inline crates/router/src/connector/payeezy/transformers.rs:45]:
```suggestion
                Err(errors::ConnectorError::NotSupported {
                message: utils::SELECTED_PAYMENT_METHOD.to_string(),
                connector: "Payeezy",
            }
```

File: crates/router/src/connector/payeezy/transformers.rs [hunk 2]
```diff
@@ -196,7 +215,21 @@ fn get_payment_method_data(
...
+        | api::PaymentMethodData::BankRedirect(_)
+        | api::PaymentMethodData::BankDebit(_)
+        | api::PaymentMethodData::BankTransfer(_)
+        | api::PaymentMethodData::Crypto(_)
+        | api::PaymentMethodData::MandatePayment
+        | api::PaymentMethodData::Reward
+        | api::PaymentMethodData::Upi(_)
+        | api::PaymentMethodData::Voucher(_)
+        | api::PaymentMethodData::GiftCard(_) => {
+            Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into())
```
Comment (srujanchikke) [inline crates/router/src/connector/payeezy/transformers.rs:231]:
```suggestion
            Err(errors::ConnectorError::NotSupported {
                message: utils::SELECTED_PAYMENT_METHOD.to_string(),
                connector: "Payeezy",
            }.into())
```

**Review by srujanchikke — CHANGES_REQUESTED** [review]
Minor changes , Other than that looks good to me!

**Comment by HeetVekariya** [comment]
@srujanchikke Thank you review, i have made the suggested changes

**Comment by swangi-kumari** [comment]
Hey @HeetVekariya ,
Pls address the CI fails
Run `cargo +nightly` fmt for formatting.

**Comment by AkshayaFoiger** [comment]
Hey @HeetVekariya, please add the default cases for `PayeezyTransactionType`

**Comment by HeetVekariya** [comment]
> Hey @HeetVekariya, please add the default cases for `PayeezyTransactionType` and map it to pending
> 
> ```
> impl ForeignFrom<(PayeezyPaymentStatus, PayeezyTransactionType)> for enums::AttemptStatus {
>     fn foreign_from((status, method): (PayeezyPaymentStatus, PayeezyTransactionType)) -> Self {
>         match status {
>             PayeezyPaymentStatus::Approved => match method {
>                 PayeezyTransactionType::Authorize => Self::Authorized,
>                 PayeezyTransactionType::Capture
>                 | PayeezyTransactionType::Purchase
>                 | PayeezyTransactionType::Recurring => Self::Charged,
>                 PayeezyTransactionType::Void => Self::Voided,
>                 _ => Self::Pending,
>             },
>             PayeezyPaymentStatus::Declined | PayeezyPaymentStatus::NotProcessed => match method {
>                 PayeezyTransactionType::Capture => Self::CaptureFailed,
>                 PayeezyTransactionType::Authorize
>                 | PayeezyTransactionType::Purchase
>                 | PayeezyTransactionType::Recurring => Self::AuthorizationFailed,
>                 PayeezyTransactionType::Void => Self::VoidFailed,
>                 _ => Self::Pending,
>             },
>         }
>     }
> }
> ```

what should i write against that default cases: error or something else

if posssible can you provide me that error message

**Comment by HeetVekariya** [comment]
@AkshayaFoiger I have added the default cases for ```PayeezyTransactionType```

**Review by swangi-kumari — APPROVED** [review]

**Review by srujanchikke — APPROVED** [review]
Looks good to me!
Thanks for the PR @HeetVekariya

**Comment by HeetVekariya** [comment]
Thank you for merging, also if i get the swag for this PR then i would like to donate it in the community

**Comment by deepanshu-iiitu** [comment]
Hey @HeetVekariya  ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Comment by HeetVekariya** [comment]
@deepanshu-iiitu I am donating this swag, i have talk with Vedant. so i will not fill this form 👍

**Pull request merged**